awscli/customizations/s3/subcommands.py

```diff
@@ -985,11 +985,18 @@ def run(self):
             'result_queue': result_queue,
         }
 
-        fgen_request_parameters = {}
+        fgen_request_parameters = {
+            'RequestPayer': self.parameters.get('request_payer'),
+        }
         fgen_head_object_params = {}
         fgen_request_parameters['HeadObject'] = fgen_head_object_params
         fgen_kwargs['request_parameters'] = fgen_request_parameters
 
+        rgen_request_parameters = {
+            'RequestPayer': self.parameters.get('request_payer'),
+        }
+        rgen_kwargs['request_parameters'] = rgen_request_parameters
```

Comment:
So for the request parameters that get passed to the `FileGenerator's`, I think I would prefer to do something more like this:
https://github.com/aws/aws-cli/blob/9cdd69c852c857f19b0e8f40caca1912e92a7675/awscli/customizations/s3/subcommands.py#L999-L1007

where we:

1) Add a key `ListObjects` to both the `fgen_request_parameters` and `rgen_request_parameters` parameters that is an empty dictionary.

2) Use the `RequestParamsMapper` class to map the `RequesterPayer` parameter into those parameter dictionaries based on if the argument was provided to the commandline.

I am mainly suggesting this because:

1) There is already precedence with how we set the `HeadObject` key in the `request_parameters`
2) It will make it easier to add future parameters for `ListObjects`.